Framework/DataHandling/src/SaveCanSAS1D2.cpp

```diff
@@ -212,19 +222,45 @@ void SaveCanSAS1D2::createSASProcessElement(std::string &sasProcess) {
   // outFile<<sasProcuserfile;
   sasProcess += sasProcuserfile;
 
+  if (m_trans_ws) {
+    // Add other run numbers
+    // SampleTransmission
+    const auto sample_trans_run =
+        getPropertyValue("SampleTransmissionRunNumber");
+    sasProcess += "\n\t\t\t<term name=\"sample_trans_run\">";
+    sasProcess += sample_trans_run + "</term>";
+
+    // SampleDirect
+    const auto sample_direct_run = getPropertyValue("SampleDirectRunNumber");
+    sasProcess += "\n\t\t\t<term name=\"sample_direct_run\">";
+    sasProcess += sample_direct_run + "</term>";
+  }
+
   // can run number if available
   if (m_transcan_ws) {
+    std::string can_run;
     if (m_transcan_ws->run().hasProperty("run_number")) {
       Kernel::Property *logP = m_transcan_ws->run().getLogData("run_number");
-      auto can_run = logP->value();
-      std::string sasProcCanRun = "\n\t\t\t<term name=\"can_trans_run\">";
-      sasProcCanRun += can_run;
-      sasProcCanRun += "</term>";
-      sasProcess += sasProcCanRun;
+      can_run = logP->value();
     } else {
       g_log.debug() << "Didn't find RunNumber log in workspace. Writing "
                        "<Run></Run> to the CANSAS file\n";
+      can_run = "";
     }
+    std::string sasProcCanRun = "\n\t\t\t<term name=\"can_trans_run\">";
+    sasProcCanRun += can_run;
+    sasProcCanRun += "</term>";
+    sasProcess += sasProcCanRun;
```

Comment:
> 9. Open up aa.xml and scroll to the bottom. There should be terms containing the run numbers in the sans table, exluding sample scatter and **can transmission**

Sample scatter isn't added, but the can transmission is present. I think it is added from the code above. I am not sure which of those is the correct behaviour. All fields in the XML:

```
<term name="sample_trans_run">42345</term>
<term name="sample_direct_run">41133</term>
<term name="can_trans_run">41137</term>
<term name="can_scatter_run">41138</term>
<term name="can_direct_run">41133</term>
```

Reply:
That's correct behaviour. In the .h5 file, sample scatter and can transmission are already present elsewhere in the file, so we only want to add the run numbers which aren't already included

Reply:
Yes that's correct behaviour, the test script was incorrect. The sample scatter is already present near the top of the XML file in a <Run> tag